test/test_fork.py

```diff
@@ -98,3 +100,42 @@ def test_topology_reset(self):
             self.assertNotEqual(child_id, init_id)
             passed, msg = parent_conn.recv()
             self.assertTrue(passed, msg)
+
+    def test_many_threaded(self):
+        # Fork randomly while doing operations.
+        class ForkThread(threading.Thread):
+            def __init__(self, runner):
+                self.runner = runner
+                super().__init__()
+
+            def run(self) -> None:
+                clients = []
+                for _ in range(10):
+                    clients.append(MongoClient())
+
+                # The sequence of actions should be somewhat reproducible.
+                # If truly random, there is a chance we never actually fork.
+                # The scheduling is somewhat random, so rely upon that.
```

Comment:
I don't understand this comment. The loop always calls fork() 3 times.

Reply:
Yeah, the ticket mentioned randomly forking, and I was initially thinking like this:
```
if random.random() <= 0.01:
   os.fork()
```
But there would be a small probability a test would exhibit behavior where it never forks, and would be less clear if it fails/passes differently than in the majority of cases. So the number and relative frequency is fixed, but the ordering of the forks is somewhat random.